From 33208e1913507a505c25877bd2f0f0c94d982e06 Mon Sep 17 00:00:00 2001
From: Fengguang Wu <fengguang.wu@intel.com>
Date: Tue, 10 Apr 2018 11:11:47 -0700
Subject: [PATCH 0341/1852] drm: xlnx: pl_disp: fix odd_ptr_err.cocci warnings

commit 853ac578d0db815f7aa0e4af4bfad199832885c5 from
https://github.com/Xilinx/linux-xlnx.git

 PTR_ERR should normally access the value just tested by IS_ERR

Generated by: scripts/coccinelle/tests/odd_ptr_err.cocci

Fixes: 742243a44a73 ("drm: xlnx: pl_disp: Use xlnx pipeline calls")
CC: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Fengguang Wu <fengguang.wu@intel.com>
Signed-off-by: Julia Lawall <julia.lawall@lip6.fr>
Reviewed-by: Saurabh Sengar <saurabhs@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_pl_disp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/xlnx/xlnx_pl_disp.c b/drivers/gpu/drm/xlnx/xlnx_pl_disp.c
index 66ee8477c26d..58a42966a6b8 100644
--- a/drivers/gpu/drm/xlnx/xlnx_pl_disp.c
+++ b/drivers/gpu/drm/xlnx/xlnx_pl_disp.c
@@ -473,7 +473,7 @@ static int xlnx_pl_disp_probe(struct platform_device *pdev)
 
 	xlnx_pl_disp->master = xlnx_drm_pipeline_init(pdev);
 	if (IS_ERR(xlnx_pl_disp->master)) {
-		ret = PTR_ERR(xlnx_pl_disp->dev);
+		ret = PTR_ERR(xlnx_pl_disp->master);
 		dev_err(dev, "failed to initialize the drm pipeline\n");
 		goto err_component;
 	}
-- 
2.31.1

